File: myPcCovid/app/src/main/java/com/giangnb/pc_covid_clone/data/room/UserDao.java
package com.giangnb.pc_covid_clone.data.room;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

@Dao
public interface UserDao {

    @Insert
    void registerUser(UserEntity userEntity);

    @Query("SELECT * FROM users where userId=(:userId) and password=(:password)")
    UserEntity login(String userId, String password);
}

File: myInstagram/app/src/main/java/com/nbgsoftware/myinstagram/core/data/database/dao/UserDAO.kt
package com.nbgsoftware.myinstagram.core.data.database.dao

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import com.base.mvvm.core.data.database.dao.BaseDAO
import com.nbgsoftware.myinstagram.core.data.database.entities.UserEntity

/**
 * Author: William Giang Nguyen | 14/8/2022
 * */
@Dao
interface UserDAO: BaseDAO<UserEntity> {

    @Query("SELECT * FROM userEntity")
    fun getAll(): List<UserEntity>

    @Insert
    fun insertAll(vararg userEntities: UserEntity)

    @Delete
    override fun delete(obj: UserEntity)

}